Flaky test: GracefulShutdownIntegrationTest.waitsForRequestToComplete

```
com.linecorp.armeria.it.server.GracefulShutdownIntegrationTest > waitsForRequestToComplete FAILED
    java.lang.AssertionError: 
    Expecting:
     <6197897120L>
    to be between:
     [5544227289L, 6144227289L]
        at com.linecorp.armeria.it.server.GracefulShutdownIntegrationTest.waitsForRequestToComplete(GracefulShutdownIntegrationTest.java:119)
```

--- a/thrift/src/test/java/com/linecorp/armeria/client/thrift/THttpClientBadSeqIdTest.java
+++ b/thrift/src/test/java/com/linecorp/armeria/client/thrift/THttpClientBadSeqIdTest.java
@@ -23,6 +23,7 @@ import java.io.InputStream;
 import java.io.OutputStream;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.net.SocketTimeoutException;
 import java.nio.charset.StandardCharsets;
 import java.util.concurrent.ExecutionException;
 
@@ -35,11 +36,14 @@ import com.linecorp.armeria.service.test.thrift.main.HelloService;
 
 public class THttpClientBadSeqIdTest {
 
-    @Test(timeout = 10000L)
+    @Test(timeout = 30000L)
     public void badSeqId() throws Exception {
         try (ServerSocket ss = new ServerSocket(0)) {
+            ss.setSoTimeout(5000);
+
             final THttpClient client = Clients.newClient(
                     "ttext+h1c://127.0.0.1:" + ss.getLocalPort(), THttpClient.class);
+
             final RpcResponse res = client.execute("/", HelloService.Iface.class, "hello", "trustin");
             assertThat(res.isDone()).isFalse();
 
@@ -67,6 +71,11 @@ public class THttpClientBadSeqIdTest {
                 while (sin.read() >= 0) {
                     continue;
                 }
+            } catch (SocketTimeoutException expected) {
+                // A connection was not accepted; Wait for the response to raise the cause.
+                res.join();
+                // Should not reach here because .join() will fail, but for completeness:
+                throw expected;
             }
 
             assertThatThrownBy(res::get)

--- a/thrift/src/test/java/com/linecorp/armeria/it/server/GracefulShutdownIntegrationTest.java
+++ b/thrift/src/test/java/com/linecorp/armeria/it/server/GracefulShutdownIntegrationTest.java
@@ -85,8 +85,8 @@ public class GracefulShutdownIntegrationTest {
         final long stopTime = System.nanoTime();
 
         // .. which should be on par with the baseline.
-        assertThat(stopTime - startTime).isBetween(baselineNanos - MILLISECONDS.toNanos(300),
-                                                   baselineNanos + MILLISECONDS.toNanos(300));
+        assertThat(stopTime - startTime).isBetween(baselineNanos - MILLISECONDS.toNanos(400),
+                                                   baselineNanos + MILLISECONDS.toNanos(400));
     }
 
     @Test(timeout = 20000L)
@@ -116,8 +116,8 @@ public class GracefulShutdownIntegrationTest {
         assertTrue(completed.get());
 
         // Should take 500 more milliseconds than the baseline.
-        assertThat(stopTime - startTime).isBetween(baselineNanos + MILLISECONDS.toNanos(200),
-                                                   baselineNanos + MILLISECONDS.toNanos(800));
+        assertThat(stopTime - startTime).isBetween(baselineNanos + MILLISECONDS.toNanos(100),
+                                                   baselineNanos + MILLISECONDS.toNanos(900));
     }
 
     @Test(timeout = 20000L)
@@ -153,8 +153,8 @@ public class GracefulShutdownIntegrationTest {
 
         // Should take 1 more second than the baseline, because the long sleep will trigger shutdown timeout.
         final long stopTime = System.nanoTime();
-        assertThat(stopTime - startTime).isBetween(baselineNanos + MILLISECONDS.toNanos(700),
-                                                   baselineNanos + MILLISECONDS.toNanos(1300));
+        assertThat(stopTime - startTime).isBetween(baselineNanos + MILLISECONDS.toNanos(600),
+                                                   baselineNanos + MILLISECONDS.toNanos(1400));
     }
 
     @Test(timeout = 20000)
@@ -182,8 +182,8 @@ public class GracefulShutdownIntegrationTest {
 
         // Should take 1 more second than the baseline, because the requests will extend the quiet period
         // until the shutdown timeout is triggered.
-        assertThat(stopFuture.join()).isBetween(baselineNanos + MILLISECONDS.toNanos(700),
-                                                baselineNanos + MILLISECONDS.toNanos(1300));
+        assertThat(stopFuture.join()).isBetween(baselineNanos + MILLISECONDS.toNanos(600),
+                                                baselineNanos + MILLISECONDS.toNanos(1400));
     }
 
     private static SleepService.Iface newClient() throws Exception {

